core/domain/certificate_assessment_domain_test.py

```
@@ -250,3 +252,392 @@ def test_from_dict_then_to_dict_matches_original_dict(self) -> None:
+    def test_validate_rejects_empty_question_id(self) -> None:
+        response = self._get_sample_response()
+        response.question_id = ''
+        with self.assertRaisesRegex(
+            Exception, 'question_id must be a non-empty string'
+        ):
+            response.validate()
+
+    def test_validate_rejects_invalid_question_version(self) -> None:
+        response = self._get_sample_response()
+        response.question_version = 0
+        with self.assertRaisesRegex(
```

Ruff (0.15.20)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)